### DISCUSSION TITLE
Filtering a built-in's table's view

### DISCUSSION THREAD

**Discussion by pierreminik** [Budibase/budibase#6104] [Help]
Hi,

I have a table where a column is an option type with three possible values, along with some meta data fields.
I need a display a Pie Chart that shows the distribution of those values, and I need some filters that allow exporing how these three values are distributed in combination of the meta data fields.

## Data

I've created a view on the table that calculates a count on the option type column and enabled `Group by` on the values. I now correctly see how the three values are distributed.

## Design

In the design, I've added the table's view as a data provider and the pie chart is shown as expected.

Now I'd like to add a filter button which filter's the original table and shows the view's calculation results. I'm only allowed to filter the view.

![Screenshot 2022-05-29 at 23 15 23](https://user-images.githubusercontent.com/614588/170900670-e422ceac-0324-4521-9e63-8f5ef7daf5dd.png)

Is this possible somehow?

**Comment by melohagan** [comment]
Hi @pierreminik 

Thank you for your post, unfortunately I believe this is a limitation of the internal table views. 

Essentially, the views are calculated at the Data layer, whereas the filter is applied at the Design layer.
I don't see there being a way to have a filter in the design layer than can affect a view in the data layer.

This use case does seem like it would be useful. I've created a feature request here: https://github.com/Budibase/budibase/issues/6108

**Edit**: If you used an external SQL table, then you could pass in bindings into a <a href="https://docs.budibase.com/docs/custom-queries">custom query</a>
